privacyidea/api/lib/utils.py

```diff
@@ -238,6 +239,32 @@ def getLowerParams(param):
     return ret
 
 
+def check_unquote(request, data):
+    """
+    Check if we need to unquote the given data.
+    Based on the user-agent header of the request we unquote the given values
+    in `data`. The user-agent string parsing is based on
+    https://httpwg.org/specs/rfc9110.html#field.user-agent
+
+    :param request: The Flask request context
+    :type request: flask.Request
+    :param data: The dictionary containing the requested data
+    :type data: dict
+    :return: New dictionary with the possibly unquoted values
+    :rtype: dict
+    """
+    # if no user agent is available, we assume that we must unquote the data
+    if not request.user_agent.string:
+        return {key: unquote(value) for (key, value) in data.items()}
+
+    ua_match = re.match(r'^(?P<agent>[a-zA-Z0-9_-]+)(/(?P<version>\d+[\d.]*)(\s.*)?)?',
+                        request.user_agent.string)
+    if ua_match and not ua_match.group('agent') in ['privacyIDEA-LDAP-Proxy']:
```

Comment:
Would you please put this in a constant at the beginning of this file near "TRUSTED_JWT_ALGOS" like

    NOT_QUOTED_USER_AGENTS = ["privacyIDEA-LDAP-Proxy"]

Reply:
Done in e173060